packages/integration/src/gitlab/core.ts

```diff
@@ -137,19 +148,27 @@ export async function getProjectId(
   }
 
   try {
-    const repo = url.pathname.split('/-/blob/')[0];
+    let repo = url.pathname.split('/-/blob/')[0];
+
+    // Get gitlab relative path
+    const relativePath = getGitLabIntegrationRelativePath(config);
+
+    // Should replace first match only
+    repo = repo.replace(relativePath, '');
```

Comment:
Should we really run this line if relativePath is empty/undefined? Also, what if the relativePath occurs in another place than the start of the string? Code like this makes me a bit worried - feels like it'd be safer to be defensive and do something similar to

```
if (relativePath && repo.startsWith(`${relativePath]/`)) {
  repo = repo.slice(relativePath.length);
}
```

Reply:
`relativePath`  is never undefined as it will always be string,
1. If it's an empty string repo name e.g `github.com` or  `mycompany.com`, will stay the same.
   * /group/repo => /group/repo
3. If the `relativePath` is set by `getGitLabIntegrationRelativePath()` to anything other than an empty string e.g `mycompany.com/github`, then we need to remove this from the repo path.
   * /gitlab/group/repo => /group/repo

 >what if the relativePath occurs in another place than the start of the string? 

As it's the relative path of the gitlab installation, we're sure it will come first before any group, repo or team as it's an integration [wide configuration](https://docs.gitlab.com/ee/install/relative_url.html#enable-relative-url-in-gitlab
), replacing the first match does make sense, but maybe I missing your question here.

Nonetheless, I will check if `relativePath` an empty string or a has a value before attempting to use it in `.replace()`